### DISCUSSION TITLE
About the problem that qEHVI and qNEHVI take too long to obtain candidate solutions

### DISCUSSION THREAD

**Discussion by LiuYang-1997** [meta-pytorch/botorch#1725] [Q&A]
In the experiment, I set up the optimization problem DTLZ7 (M=5, D=6, M is the number of objectives, and D is the number of decision variables). I set batch_size = 5, and found that one run takes a long time (sometimes it takes 30 minutes), what is the reason?

**Comment by saitcakmak** [comment]
Hi @LiuYang-1997. Can you provide more details? There's very little information here for me to help you with. Regarding batch size, unless you're using sequentially greedy optimization (passing `sequential=True` to `optimize_acqf`) the cost of qNEHVI will increase exponentially in `batch_size` (`q` in `optimize_acqf`).

**Reply by sdaulton** [reply]
+1 to Sait’s comment. Be sure to use sequential greedy batch optimization. The box decomposition algorithm will also be expensive with 5 objectives. If you are using qNEHVI, a box decomposition is computed for each MC sample, so you can speed things up by reducing the number of mc samples. Each box decomposition can also contain a lot boxes, and since hypervolume improvement can be computed in a highly parallel fashion, including over boxes, using a GPU Will speed things up quite a bit.

**Reply by LiuYang-1997** [reply]
ok, I will try this. Thank you very much.
